lib/req/steps.ex

```diff
@@ -36,14 +36,28 @@ defmodule Req.Steps do
   def put_base_url(request)
 
   def put_base_url(%{options: %{base_url: base_url}} = request) do
-    url = URI.merge(base_url, request.url)
-    %{request | url: url}
+    if request.url.scheme != nil do
+      # skip if url is already with scheme
+      request
+    else
+      %{request | url: URI.parse(join(base_url, request.url))}
+    end
   end
 
   def put_base_url(request) do
     request
   end
 
+  defp join(base, url) do
+    case {:binary.last(base), to_string(url)} do
+      {"/", "/" <> rest} -> base <> rest
```

Comment:
we're missing some tests because this logic is wrong. Could you look into missing tests?

```
iex> :binary.last("foo/")
47
```

that's why I mentioned we need to match on `?/`, the `/` char code:

```
iex> ?/
47
```

Reply:
Thanks, I was not aware that `?/` was the charcode for `/`. That was why that seems a bit cryptic to me.

But seems like the tests are failing because of connection issues. I am not able to reproduce locally.

Also am not able to understand how `test/req/steps_test.exs:1019`, was affected by this change. The request does not have a `base_url`, so might be a flaky one?

Reply:
Yes it's a flaky test, don't worry about it.

As I mentioned earlier, we had a wrong logic (`:binary.last` + `"/"`) that was not caught by any of the new tests so I'd appreciate it if you could investigate it further and either add more tests, change the logic, or bit of both.

Reply:
Sorry, I feel so stupid for not seeing that. So thanks for keeping an eye on me.

Seems like Bypass accepts a path like `/api/v2//foo` when I specify a path like `/api/v2/foo`. But doing an assert that checks `conn.request_path` would have caught the bug I made with `:binary.last` + `"/"`.